app/src/ui/repositories-list/group-repositories.ts

```diff
@@ -44,12 +45,22 @@ export function groupRepositories(repositories: ReadonlyArray<Repositoryish>): R
     const repositories = grouped.get(identifier)
     if (!repositories || repositories.length === 0) { return }
 
+    const names = new Map<string, number>()
+    for (const repository of repositories) {
+      const existingCount = names.get(repository.name) || 0
+      names.set(repository.name, existingCount + 1)
+    }
+
     repositories.sort((x, y) => caseInsensitiveCompare(x.name, y.name))
-    const items = repositories.map(r => ({
-      text: r.name,
-      id: r.id.toString(),
-      repository: r,
-    }))
+    const items: ReadonlyArray<IRepositoryListItem> = repositories.map(r => {
```

Comment:
We _can_, but having the type annotation makes the compile errors much clearer when we change `IRepositoryListItem`.